src/core_plugins/kibana/public/home/components/tutorial/tutorial.js

```diff
@@ -104,6 +117,50 @@ export class Tutorial extends React.Component {
     });
   }
 
+  checkInstructionSetStatus = async (instructionSetIndex) => {
+    const instructions = this.getInstructions();
+    const esHitsCheckConfig = _.get(instructions, `instructionSets[${instructionSetIndex}].statusCheck.esHitsCheck`);
+    const { hasFailed, isComplete } = await this.fetchEsHitsStatus(esHitsCheckConfig);
+
+    this.setState((prevState) => {
+      const statusCheck = _.cloneDeep(prevState.statusCheck);
+      statusCheck[instructionSetIndex].isComplete = isComplete;
+      statusCheck[instructionSetIndex].hasFailed = hasFailed;
+      statusCheck[instructionSetIndex].isFetchingStatus = false;
+      return { statusCheck };
+    });
+  }
+
+  fetchEsHitsStatus = async (esHitsCheckConfig) => {
+    if (!esHitsCheckConfig) {
+      return { hasFailed: false, isComplete: false };
+    }
+
+    const searchHeader = JSON.stringify({ index: esHitsCheckConfig.index });
+    const searchBody = JSON.stringify({ query: esHitsCheckConfig.query, size: 1 });
+    const response = await fetch(this.props.addBasePath('/elasticsearch/_msearch'), {
+      method: 'post',
+      body: `${searchHeader}\n${searchBody}\n`,
+      headers: {
+        accept: 'application/json',
+        'content-type': 'application/x-ndjson',
+        'kbn-xsrf': 'kibana',
+      },
+      credentials: 'same-origin'
+    });
+
+    if (response.status > 300) {
+      return { hasFailed: true, isComplete: false };
```

Comment:
This also pushes me to think an enum would be easier to follow.  I thought this would be a bad state, but I guess you are using incomplete plus failed to mean there was an error, rather than a successful check but with no data?  Below it's also a bit confusing that it's broken into two states that will always be mutually exclusive.